app/models/role.rb

```diff
@@ -41,7 +41,7 @@ def role_name
   end
 
   replace_scope :system, -> do
-    not_system.where(system: true)
```

Comment:
This was the root cause of the bug since it resulted in `where system = false and system = true`.